### DISCUSSION TITLE
WS-SecurityPolicy

### DISCUSSION THREAD

**Discussion by brett-beaumont** [quarkiverse/quarkus-cxf#310] [General]
Does this extension support WS-SecurityPolicy? If so, do you have any hints for how to set this up? I have the WSS4J interceptor enabled and running to check signatures, but I really want the rest of policy enabled as well - for example, checking that the whole message is signed. 

Thanks,
Brett

**Comment by shumonsharif** [comment]
Hi Brett, yes, WS-SecurityPolicy should work, and what you're describing can be easily configured as per the docs using the `DefaultCryptoCoverageChecker` interceptor.

https://quarkiverse.github.io/quarkiverse-docs/quarkus-cxf/dev/ws-security.html
https://cxf.apache.org/docs/ws-security.html

Is there a specific issue you're running into?  If so, please feel free to share a reproducer we can look at.
